src/parser.ml

```diff
@@ -470,7 +561,7 @@ let entity s =
             loop (succ m) (n * 16 + Char.code c - Char.code '0') (Sub.tail s)
         | Some ';' ->
             if m = 0 then raise Fail;
-            let u = if n = 0 || not (Uchar.is_valid n) then Uchar.rep else Uchar.of_int n in
+            let u = if n = 0 || not (Uchar.is_valid n) then Compat.rep else Uchar.of_int n in
```

Comment:
This change (and the ones below) are not needed if we organize `Compat` as suggested above.